mitodl/open-discussions
# Allow article cover images to be deleted

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1691

#### What's this PR do?
- Removes an article cover image if `article_cover` is not sent with a patch request.
- Prevents an exception from occurring if `article_cover` in the patch request is a URL instead of an uploaded file (which happens on master when editing an existing article but leaving the cover as is).

#### How should this be manually tested?
- Edit an article with a cover image.  Change the text but not the cover image.  Save.  You should not see an exception when the post detail page reloads, and the cover image should still exist.
- Edit the article again.  Remove the cover image.  Save.  You should not see an exception when the post detail page reloads, and the cover image should be gone.
